docs/src/index.md

```diff
@@ -2,13 +2,179 @@
+  and undo the transformation without having to do all the manual work
+  keeping constants around.
+
+- Pipelines can be easily constructed with clean syntax
+  `(f1 → f2 → f3) ⊔ (f4 → f5)`, and they are automatically
+  revertible when the individual transforms are revertible.
+
+- Branches of a pipeline and colwise transforms are run in parallel
+  using multiple threads with the awesome
+  [Transducers.jl](https://github.com/JuliaFolds/Transducers.jl)
+  framework.
+
+- Pipelines can be reapplied to unseen "test" data using the same cache
+  (e.g. constants) fitted with "training" data. For example, a `ZScore`
+  relies on "fitting" `μ` and `σ` once at training time.
+
+## Rationale
+
+A common task in statistics and machine learning consists of transforming
+the variables of a problem to achieve better convergence or to apply methods
+that rely on multivariate Gaussian distributions. This process can be quite
+tedious to implement by hand and very error-prone. We provide a consistent
+and clean API to combine statistical transforms into pipelines.
+
+*Although most transforms discussed here come from the statistical domain,
+our long term vision is more ambitious. We aim to provide a complete
+user experience with fully-featured pipelines that include standardization
+of column names, imputation of missing data, and more.*
+
+## Installation
+
+Get the latest stable release with Julia's package manager:
+
+```julia
+] add TableTransforms
+```
+
+## Usage
+
+Below is a quick example with simple transforms:
+
+```@example usage
+using TableTransforms
+using Plots, PairPlots
+using Distributions
+using Random; Random.seed!(2) # hide
+gr(format=:png) # hide
+
+# example table from PairPlots.jl
+N = 100_000
+a = [2randn(N÷2) .+ 6; randn(N÷2)]
+b = [3randn(N÷2); 2randn(N÷2)]
+c = randn(N)
+d = c .+ 0.6randn(N)
+table = (; a, b, c, d)
+
+# corner plot of original table
+table |> corner
+```
+
+```@example usage
+# convert to PCA scores
+table |> PCA() |> corner
+```
+
+```@example usage
+# convert to any Distributions.jl
+table |> Quantile(Normal()) |> corner
+```
+
+Below is a more sophisticated example with a pipeline that has
+two parallel branches. The tables produced by these two branches
+are concatenated horizontally in the final table:
+
+```@example usage
+# create a transform pipeline
+f1 = ZScore()
+f2 = Scale()
+f3 = Quantile()
+f4 = Functional(cos)
+f5 = Interquartile()
+pipeline = (f1 → f2 → f3) ⊔ (f4 → f5)
+
+# feed data into the pipeline
+table |> pipeline |> corner
 ```
 
-```@autodocs
-Modules = [TableTransforms]
+To create sequential pipeline you can use `→` (`\to<tab>`) operator and
+to crate parallel pipeline you can use `⊔` (`\sqcup<tab>`) operator:
+
+```@docs
+→
+⊔
+```
+
+### Apply and Revert
+
+To revert a pipeline or single transform, use the `apply` and `revert`
+functions instead:
+
+```@docs
+apply
+revert
+isrevertible
+```
+
+To exemplify the use of these functions, let's create a table:
+
+```@example usage2
```

Comment:
No need to create a separate usage2. Just reuse the imports from usage, it will clean up the example. Also, I think we can drop the DataFrames.jl dependency from the docs and use small named tuples to clarify the idea.

Reply:
I used DataFrames because it has an HTML view of the table, but I can use named tuples and PrettyTables.jl instead.

Reply:
I think a small named tuple like (a=[1,2,3], b=[4,5,6], c=[7,8,9]) or similar will be enough, no need for fancy printing. We just need to illustrate the reversibility and readers can do the checks quickly with these small number of values.